__tests__/app.test.js

```diff
@@ -16,51 +16,151 @@ beforeEach(() => {
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        },
+      };
+      return request(app)
+        .get("/api/articles/1")
+        .then(({ body }) => {
+          expect(body).toMatchObject(articleTemplate);
         });
-    })
-})
+    });
+    test("Should return status 404 and error message if article_id does not exist", () => {
+      return request(app)
+        .get("/api/articles/999")
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe(`We can't find that!`);
+        });
+    });
+    test("Should return status 400 and error message if article_id is invalid type", () => {
+      return request(app)
+        .get("/api/articles/bananaPancakes")
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe(`That input is invalid`);
+        });
+    });
+  });
+  describe("POST /api/articles/:article_id/comments", () => {
+          const commentThis = {
+            username: "butter_bridge",
+            body: `Did you just finish all the coffee?!`,
+          }; 
+          const matchedShape = {
+            comment_id: 19,
+            body: `Did you just finish all the coffee?!`,
+            article_id: 3,
+            author: "butter_bridge",
+            votes: 0,
+            created_at: '2023-08-16T17:57:37.000Z',
+          };
+    test("Should return status 201 when new comment successfully posted", () => {
+      return request(app)
+        .post("/api/articles/3/comments")
+        .send({ ...commentThis })
+        .then(() => {
+          expect(201);
+        });
+    });
+    test("Should return posted comment showing all desired keys", () => {
+      return request(app)
+        .post("/api/articles/3/comments")
+        .send({ ...commentThis })
+        .then(({ body }) => {
+          expect(Object.keys(body)).toEqual(Object.keys({...matchedShape}));
+        });
+    });
+    test("Should return 400 and error message if user attempts to post empty body", () => {
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send({ username: "Monstera_Munch" })
+        .then(({ body: { msg } }) => {
+          expect(400)
+          expect(msg).toBe(`Nothing to post!`);
+        });
+    });
+    test("Should return 400 and error message if user attempts to post anonymously", () => {
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send({ body: "The mystery poster strikes again!" })
+        .then(({ body: { msg } }) => {
+          expect(400);
+          expect(msg).toBe(`No anonymous posting allowed, please log in`);
+        });
+    });
+    test("Should return 400 and error message if request body has undesired keys", () => {
+      return request(app)
+        .post("/api/articles/1/comments")
```

Comment:
This is really good as a thing to think about for your API, if you want your API to be expandable in the future without breaking your tests... do we want extra keys to give an error ? 
Let's say that down the line we want to also be able to add something extra to the POST body, would it be better for the endpoint to just ignore unnecessary keys ?